### DISCUSSION TITLE
Problem with binding to ClusterIP

### DISCUSSION THREAD

**Discussion by khramov86** [strongswan/strongswan#303] [General]
While trying to make work HA with strongswan faced with problem/misunderstanding.
My `ipsec.conf` file
```
config setup
conn %default
  rekey=yes
  reauth=yes
  ike=aes128-sha256-modp1536!
  esp=aes128-sha256-modp1536!
conn testconn
  authby=secret
  auto=start
  leftid=128.0.26.15 #this is my virtualip
  left=128.0.26.15
  leftsubnet=10.9.0.0/24
  rightid=128.0.26.16
  right=128.0.26.16
  rightsubnet=10.8.0.0/24
  keyexchange=ikev2
  mobike=no
  ikelifetime=60m
  keylife=20m
  rekeymargin=3m
  keyingtries=1
  keyexchange=ikev2
```
my `strongswan.conf` file
```
charon {
        load_modular = yes
        plugins {
                include strongswan.d/charon/*.conf
        ha {
               local = 10.9.0.120
               remote = 10.9.0.121
               segment_count = 2
               fifo_interface = yes
               monitor = yes
               autobalance = 10
                }
        }
}
```
Added to iptables CLUSTERIP
```
-A INPUT -d 128.0.26.15-i eth1 -j CLUSTERIP --new --hashmode sourceip --clustermac 01:00:C0:A8:00:05 --total-nodes 2 --local-node 1 --hash-init 0
-A INPUT -d 10.9.0.162 -i eth0 -j CLUSTERIP --new --hashmode sourceip --clustermac 01:00:0A:01:00:05 --total-nodes 2 --local-node 1 --hash-init 0
```
While trying to start strongswan service - getting this error:
```
Apr  5 10:35:22 test-ipsec1 charon: 10[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
Apr  5 10:35:22 test-ipsec1 charon: 10[NET] sending packet: from 128.0.26.15[500] to 128.0.26.16[500] (400 bytes)
Apr  5 10:35:22 test-ipsec1 charon: 04[NET] error writing to socket: Network is unreachable
```
So strongswan can't bind to CLUSTERIP. 
Can you help me with this?

**Comment by Thermi** [comment]
You need to bind the ip address to a local interface. The iptables CLUSTERIP target is just for filtering the packets. It doesn't bind any IPs.

**Reply by Thermi** [reply]
Did that work out for you? If so, please mark the answer as accepted.

**Reply by khramov86** [reply]
Still trying to make HA work, but workaround with assigning address to interface resolved bind issue. Thank you!

**Reply by khramov86** [reply]
@Thermi Can you please, give an advice? When i'm trying to start second node - getting in logs:
```
Apr 13 17:54:20 test-ipsec2 charon: 07[CFG] HA segment 1 was handled twice, taking
Apr 13 17:54:21 test-ipsec2 charon: 07[CFG] HA segment 1 was handled twice, taking
```

Also getting 
```
Apr 13 18:11:41 test-ipsec2 charon: 05[CFG] writing to CLUSTERIP file '/proc/net/ipt_CLUSTERIP/lan_vip' failed: Cannot allocate memory
Apr 13 18:11:41 test-ipsec2 charon: 05[CFG] writing to CLUSTERIP file '/proc/net/ipt_CLUSTERIP/wan_vip' failed: Cannot allocate memory
```

**Reply by Thermi** [reply]
That is not a workaround, that is the proper way to do it.
You need to specify a different segment on each node. I do not know about the internals of the CLUSTERIP target or the cluster match. Try using the cluster match module instead. The CLUSTERIP target is deprecated. It might work better. Other than that, I'd need to review the source code of it to tell you exactly what is wrong.

**Reply by khramov86** [reply]
Thank you, Thermi. You advice helped me to solve problem.
Also there was a mistake in my config with cluster ip, they didn't work properly without netmask `/32`.
